(go1.20rc1) reflect: call of reflect.Value.Int on struct Value

```
--- FAIL: TestRWMutexLocked (0.00s)
panic: reflect: call of reflect.Value.Int on struct Value [recovered]
        panic: reflect: call of reflect.Value.Int on struct Value

goroutine 7 [running]:
testing.tRunner.func1.2({0x50b9a0, 0xc0000100f0})
        /home/preston/sdk/gotip/src/testing/testing.go:1526 +0x24e
testing.tRunner.func1()
        /home/preston/sdk/gotip/src/testing/testing.go:1529 +0x39f
panic({0x50b9a0, 0xc0000100f0})
        /home/preston/sdk/gotip/src/runtime/panic.go:884 +0x213
reflect.Value.Int(...)
        /home/preston/sdk/gotip/src/reflect/value.go:1461
github.com/trailofbits/go-mutexasserts.RWMutexRLocked(0xc00004e760?)
        /tmp/go-mutexasserts/mutex.go:22 +0x13d
github.com/trailofbits/go-mutexasserts.TestRWMutexLocked(0xc000007ba0)
        /tmp/go-mutexasserts/asserts_test.go:29 +0x45
testing.tRunner(0xc000007ba0, 0x5355a8)
        /home/preston/sdk/gotip/src/testing/testing.go:1576 +0x10b
created by testing.(*T).Run
        /home/preston/sdk/gotip/src/testing/testing.go:1629 +0x3ea
FAIL    github.com/trailofbits/go-mutexasserts  0.004s
FAIL
```

Using gotip version
```
go version devel go1.20-db36eca Thu Dec 29 23:36:05 2022 +0000 linux/amd64
```

To repro, use gotip or go1.20rc1
```
gotip test -tags debug ./...
```

--- a/mutex.go
+++ b/mutex.go
@@ -19,5 +19,5 @@ func RWMutexLocked(rw *sync.RWMutex) bool {
 }
 
 func RWMutexRLocked(rw *sync.RWMutex) bool {
-	return reflect.ValueOf(rw).Elem().FieldByName("readerCount").Int() > 0
+	return readerCount(rw) > 0
 }

--- a/readercount_go120.go
+++ b/readercount_go120.go
@@ -1,0 +1,18 @@
+//go:build go1.20
+// +build go1.20
+
+package mutexasserts
+
+import (
+	"reflect"
+	"sync"
+	"sync/atomic"
+)
+
+// Starting in go1.20, readerCount is an atomic int32 value.
+// See: https://go-review.googlesource.com/c/go/+/429767
+func readerCount(rw *sync.RWMutex) int64 {
+	// Look up the address of the readerCount field and use it to create a pointer to an atomic.Int32,
+	// then load the value to return.
+	return int64((*atomic.Int32)(reflect.ValueOf(rw).Elem().FieldByName("readerCount").Addr().UnsafePointer()).Load())
+}

--- a/readercount_go120.go
+++ b/readercount_go120.go
@@ -14,5 +14,6 @@ import (
 func readerCount(rw *sync.RWMutex) int64 {
 	// Look up the address of the readerCount field and use it to create a pointer to an atomic.Int32,
 	// then load the value to return.
-	return int64((*atomic.Int32)(reflect.ValueOf(rw).Elem().FieldByName("readerCount").Addr().UnsafePointer()).Load())
+	rc := (*atomic.Int32)(reflect.ValueOf(rw).Elem().FieldByName("readerCount").Addr().UnsafePointer())
+	return int64(rc.Load())
 }

--- a/readercount_legacy.go
+++ b/readercount_legacy.go
@@ -1,0 +1,14 @@
+//go:build !go1.20
+// +build !go1.20
+
+package mutexasserts
+
+import (
+	"reflect"
+	"sync"
+)
+
+// Prior to go1.20, readerCount was an int value.
+func readerCount(rw *sync.RWMutex) int64 {
+	return reflect.ValueOf(rw).Elem().FieldByName("readerCount").Int()
+}

